requesttojson/requesttojson.go

```diff
@@ -0,0 +1,58 @@
+package requesttojson
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+// Struct to represent the HTTP request in JSON format
+type RequestJSON struct {
+	Method  string              `json:"method"`
+	URL     string              `json:"url"`
+	Proto   string              `json:"proto"`
+	Headers map[string][]string `json:"headers"`
+	Body    string              `json:"body"`
+	Query   map[string][]string `json:"query_parameters"`
+}
+
+func RequestToJSON(req *http.Request, replaceEmptyBodyWith string) (json.RawMessage, RequestJSON, error) {
+	defer func() {
+		_ = req.Body.Close()
```

Comment:
Should we check for nil as well?
```suggestion
		if req.Body != nil {
			_ = req.Body.Close()
		}
```